utils/nlp.py

```diff
@@ -32,3 +32,22 @@ def lowercase_strip_punc(self, document):
             char for char in self.normalize(document)
             if not unicodedata.category(char)[0] in self.punct
         )
+
+    def clean_str(self, document):
```

Comment:
This is a great candidate for a unit test. I'm not sure what exactly all of these transformations are, and a number of different test case examples would be very helpful.